### PR TITLE
Feat/authjs

### PR DESCRIPTION
**Pull request by takoyaro**

### CODE REVIEW

File: modules/odr_frontend/src/routes/admin/users/api/+server.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { pgClient, type IDBUser } from '$lib/server/pg';
+import type { RequestHandler } from '@sveltejs/kit';
+export const PUT: RequestHandler = async ({ request }) => {
+	const body = await request.json();
+	if (!body.user) {
+		return new Response(JSON.stringify({ success: false, error: 'No user provided' }));
+	}
+	const user: IDBUser = body.user;
+	try {
+		const result = await pgClient.query('UPDATE users SET is_superuser=$1 WHERE id=$2', [
```
Comment (Tomislaw) [inline modules/odr_frontend/src/routes/admin/users/api/+server.ts:10]: Is there any reason we don't want to use custom backend endpoint for that?

File: modules/odr_frontend/src/lib/server/pg.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	connectionTimeoutMillis: 2000
+};
+
+export const pgClient = new pg.Client(pg_client_config);
+
+// Connect the client once when the module is loaded
+pgClient.connect();
+
+export const PG_API = {
+	users: {
```
Comment (Tomislaw) [inline modules/odr_frontend/src/lib/server/pg.ts:45]: We have endpoints for most of these operations on backend side.

The issue with having postgres calls on both frontend and backed is that if we find any issue with our workflow, we have to update it on both sides.

I know that you probably didn't want to wait for other team members to implement these changes which were needed for your feature and I also think we should do it the fast way to keep things running
But IMO it is something we should fix, at least in the future.

File: modules/odr_frontend/src/routes/+layout.svelte [hunk 3]
```diff
@@ -23,11 +24,25 @@
 			<strong class="text-xl uppercase"><a href="/">OMI DATA PIPELINE</a></strong>
 		</svelte:fragment>
 		<svelte:fragment slot="trail">
-			<a href="/datasets" class="btn btn-sm">Request Dump</a>
-			{#if !data.isAuthenticated}
-				<a href="/auth/login" class="btn btn-sm">Login</a>
+			{#if !$page.data.session?.user}
+				<span class="btn btn-sm">Sign In:</span>
+				<button
```
Comment (CheesyLaZanya) [inline modules/odr_frontend/src/routes/+layout.svelte:29]: Should there be a GitHub sign in icon here as well?

**Review by CheesyLaZanya — APPROVED** [review]
I created a number of issues to track work I think must follow this, either minor issues or follow up work to get us across the line on the new work, specifically #65 , #66 , #67 , #68 , #69 , #70 , #71 , #73, #74, #75

Other than those items, I was able to test authentication with both GitHub and Discord, the admin page only being accessible with super user access, adding teams, and adding users to teams.

**Comment by fearnworks** [comment]
Can we resolve #75  before merging?

**Review by Tomislaw — APPROVED** [review]

**Review by fearnworks — APPROVED** [review]

**Pull request merged**